src/abstracttypes.jl

```diff
@@ -0,0 +1,31 @@
+"""
+    AbstractHypergraph{T} <: AbstractMatrix{Union{T, Nothing}}
+
+An abstract hypergraph type storing information about vertices and hyperedges.
+"""
+# TODO: should this be just `abstract type AbstractHypergraph{T} end`?
+# How important is subtyping AbstractMatrix?
```

Comment:
subtyping AbstractMatrix was given so the hypergraph properties can be easily accessed and manipulater. 
You can do  `hg[4,1] = 3` to assign weight 3 to the relation between vertex 4 and hyperedge 1. 
Changing this would be breaking for everyone